File: bibliography.py
import json
import requests

zotero_api_url = "https://api.zotero.org/groups/4724580/items?format=csljson"

def gen_id(zotero_item):
    first_author_family = zotero_item['author'][0]['family']
    year = str(zotero_item['issued']['date-parts'][0][0])
    return first_author_family + year

with requests.Session() as s:
    download = s.get(zotero_api_url)
    decoded_content = download.content.decode('utf-8')

    zotero_data = json.loads(decoded_content)
    zotero_items = zotero_data['items']
    for i, zotero_item in enumerate(zotero_items):
        zotero_items[i]['id'] = gen_id(zotero_item)

    json_bib = json.dumps(zotero_items, indent=4, ensure_ascii=False)
    with open('./assets/bib.json', "w") as f:
        f.write(json_bib)

File: bibliography.js
const fs = require("fs");
const path = require("path");

const url = "https://api.zotero.org/groups/2660172/items?format=csljson";

function genId({ author: [{ family }], issued }) {
  const year = String(issued["date-parts"][0][0]);
  return family + year;
}

fetch(url)
  .then(async (response) => {
    const data = await response.json();

    console.log(`${data.items.length} bibliographic items founded`);
    
    data.items = data.items.map((item) => {
      item.id = genId(item);
      return item;
    });
    fs.writeFile(
      path.join(__dirname, "assets", "bib.json"),
      JSON.stringify(data.items, undefined, 2),
      "utf-8",
      (err) => err && console.error("Unable to write file:", err)
    );
  })
  .catch((err) => {
    console.error("Unable to fetch:", err);
  });
